pennylane/data/__init__.py

```diff
@@ -133,6 +133,9 @@
 >>> my_dataset = Dataset.open("~/dataset/my_dataset/h5", mode="copy")
 >>> my_dataset.new_attribute = "abc"
 
+Since opened datasets stream data from the disk, it is not possible to simultaneously access the same dataset
```

Comment:
Please consider highlighting this information by creating a Note. It seems that several users have faced this issue.

Reply:
Rather than a note, I would use important or warning!

Reply:
Updated with `.. important::`